src/components/panel/panel.js

```diff
@@ -1160,3 +1239,131 @@ MdPanelPosition.prototype.right = function(opt_right) {
 MdPanelPosition.prototype.getRight = function() {
   return this._right;
 };
+
+
+
+/*****************************************************************************
+ *                               MdPanelAnimation                            *
+ *****************************************************************************/
+
+
+/**
+ * Animation configuration object. To use, create an MdPanelAnimation with the
+ * desired properties, then pass the object as part of $mdPanel creation.
+ *
+ * Example:
+ *
+ * var panelAnimation = new MdPanelAnimation()
+ *     .openFrom(myButtonEl)
+ *     .closeTo('.my-button')
+ *     .withAnimation('');
+ *
+ * $mdPanel.create({
+ *   animation: panelAnimation
+ * });
+ *
+ * @final @constructor
+ */
+function MdPanelAnimation() {
+  /** @private {!Object} */
+  this._openFrom = {};
+
+  /** @private {!Object} */
+  this._closeTo = {};
+}
+
+
+/**
+ * Gets the boundingClientRect for the opening animation.
+ * @returns {!Object}
+ */
+MdPanelAnimation.prototype.getOpenFrom = function() {
+  return this._openFrom;
+};
+
+
+/**
+ * Gets the boundingClientRect for the closing animation.
+ * @returns {!Object}
```

Comment:
The bottom and right aren't always there. If you need them for positioning,
we can add it in that CL.

On Thu, Mar 24, 2016 at 4:10 PM Erin Coughlan <notifications@github.com>
wrote:

> In src/components/panel/panel.js
> <https://github.com/angular/material/pull/7718#discussion_r57401883>:
>
> > +  this._closeTo = {};
> > +}
> > +
> > +
> > +/**
> > + * Gets the boundingClientRect for the opening animation.
> > + * @returns {!Object}
> > + */
> > +MdPanelAnimation.prototype.getOpenFrom = function() {
> > +  return this._openFrom;
> > +};
> > +
> > +
> > +/**
> > + * Gets the boundingClientRect for the closing animation.
> > + * @returns {!Object}
>
> It also has bottom and right, which I will be using in the relative
> positioning CL. Should be make a typedef like thing somewhere in this file
> to specify it?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/angular/material/pull/7718/files/7ab60d5adb0603b82da8b7ecaa9aefa4b74af36b#r57401883>
>